zapcore/json_encoder.go

```diff
@@ -60,10 +52,10 @@ type jsonEncoder struct {
 // libraries will ignore duplicate key-value pairs (typically keeping the last
 // pair) when unmarshaling, but users should attempt to avoid adding duplicate
 // keys.
-func NewJSONEncoder(cfg JSONConfig) Encoder {
+func NewJSONEncoder(cfg EncoderConfig) Encoder {
```

Comment:
I guess it's mildly weird that we store a pointer to a passed value, rather than letting the caller pass a pointer (e.g. so that multiple json-encoded facilities can share the same config). Doubly so if this new config achieves it's namesake of being the config for all encoders, for sharing b/w console and json.